app/uk/gov/hmrc/tradergoodsprofilesrouter/service/RouterService.scala

```diff
@@ -200,6 +206,107 @@ class RouterServiceImpl @Inject() (eisConnector: EISConnector, uuidService: Uuid
     )
   }
 
+  override def createRecord(request: CreateRecordRequest)(implicit
+    ec: ExecutionContext,
+    hc: HeaderCarrier
+  ): EitherT[Future, Result, CreateRecordResponse] = {
+    val correlationId = uuidService.uuid
+    EitherT(
+      eisConnector
+        .createRecord(request, correlationId)
+        .map(result => Right(result))
+        .recover {
+          case UpstreamErrorResponse(message, BAD_REQUEST, _, _) =>
+            Left(BadRequest(Json.toJson(determine400Error(correlationId, message))))
+
+          case UpstreamErrorResponse(_, FORBIDDEN, _, _) =>
+            Left(
```

Comment:
A lot of these error responses are duplicated, by the fetch record. They could be put into their own functions and pass in the correlationId

Reply:
That's true, but we agreed to refactor it at a later time. If you notice, they are also duplicated in `fetchRecords`. So it's best to do the refactor when we have each of their unique cases known. cc @AmarkumarJagtap

Reply:
I thought we were refactor later for the GetSingle Record router, but I think now it is the moment we should refactor as the code is getting to confusing especially the GetRecord and GetRecords as there is a lot of duplication on the RouteService. I have started doing that in the last couple of dau in parallel wit the GetRecord external API. We can do this an exercise if you want

Reply:
That's alright then. I will refactor this now as part of this PR